src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceServiceTest.kt

```diff
@@ -1351,7 +1351,255 @@ class LicenceServiceTest {
       "2",
     )
   }
+  @Test
+  fun `should set the license status to inactive when the offender has a new future conditional release date`() {
+    whenever(licenceRepository.findById(1L)).thenReturn(Optional.of(aLicenceEntity.copy(statusCode = LicenceStatus.ACTIVE)))
+    whenever(prisonApiClient.hdcStatus(any())).thenReturn(
+      Mono.just(
+        PrisonerHdcStatus(
+          approvalStatusDate = null, approvalStatus = "REJECTED", refusedReason = null,
+          checksPassedDate = null, bookingId = aLicenceEntity.bookingId!!, passed = true
+        )
+      )
+    )
+    service.updateSentenceDates(
+      1L,
+      UpdateSentenceDatesRequest(
+        conditionalReleaseDate = LocalDate.now().plusDays(5),
+        actualReleaseDate = null,
+        sentenceStartDate = LocalDate.parse("2018-10-22"),
+        sentenceEndDate = LocalDate.parse("2024-09-11"),
+        licenceStartDate = LocalDate.parse("2023-09-11"),
+        licenceExpiryDate = LocalDate.parse("2024-09-11"),
+        topupSupervisionStartDate = LocalDate.parse("2024-09-11"),
+        topupSupervisionExpiryDate = null,
+      )
+    )
+
+    val licenceCaptor = ArgumentCaptor.forClass(EntityLicence::class.java)
+    val auditCaptor = ArgumentCaptor.forClass(EntityAuditEvent::class.java)
```

Comment:
there's some [nice syntax](https://github.com/mockito/mockito-kotlin/tree/a6f860461233ba92c7730dd42b0faf9ba2ce9281#argument-captors) for doing this in kotlin mockito.

Though looks like this is in keeping with how its done with other tests in this PR so probably not really worth changing

Reply:
Thank you Andrew for point it out.  Since I am new to Kotlin , I used existing test as a reference. But I will bear that in my for future